src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -224,9 +230,26 @@ public Future<Void> updateOrder(CompositePurchaseOrder compPO, boolean deleteHol
       .map(HelperUtils::convertToCompositePurchaseOrder)
       .compose(lines -> purchaseOrderLineService.populateOrderLines(lines, requestContext))
       .compose(poFromStorage -> {
+        CompositePurchaseOrder clonedPoFromStorage = JsonObject.mapFrom(poFromStorage).mapTo(CompositePurchaseOrder.class);
         boolean isTransitionToOpen = isTransitionToOpen(poFromStorage, compPO);
         return orderValidationService.validateOrderForUpdate(compPO, poFromStorage, deleteHoldings, requestContext)
-          .compose(ok -> {
+          .compose(v -> {
+            var poLineFutures = new ArrayList<Future<Void>>();
+            if (!compPO.getCompositePoLines().isEmpty()) {
+              compPO.getCompositePoLines().forEach(poLine -> {
+                var compPoLineFromStorage = clonedPoFromStorage.getCompositePoLines().stream()
+                  .filter(Objects::nonNull).filter(entry-> Objects.nonNull(entry.getId()))
+                  .filter(entry -> entry.getId().equals(poLine.getId())).findFirst().orElse(null);
```

Comment:
StringUtils.equals can be used to find correct po lnie from storage, so in this case 1 filter will be used and this method also avoids NPE